src/collections/componentSet.ts

```diff
@@ -276,12 +276,15 @@ export class ComponentSet extends LazyCollection<MetadataComponent> {
           typeMap.set(typeName, []);
         }
         const typeEntry = typeMap.get(typeName);
-        if (fullName === ComponentSet.WILDCARD) {
+        if (fullName === ComponentSet.WILDCARD && !type.supportsWildcardAndName) {
+          // if the type doesn't support mixed wildcards and specific names, overwrite the names to be a wildcard
           typeMap.set(typeName, [fullName]);
-        } else {
-          if (!typeEntry.includes(fullName) && !typeEntry.includes(ComponentSet.WILDCARD)) {
-            typeMap.get(typeName).push(fullName);
-          }
+        } else if (
+          type.supportsWildcardAndName ||
```

Comment:
the original code 
```ts
if (!typeEntry.includes(fullName) && !typeEntry.includes(ComponentSet.WILDCARD))
```
was preventing duplicates, right?
with the `||` condition, if type === customObject (supportsWildcardAndName), you'd be allowing duplicates, right?

Reply:
you're right!! I'm not sure if a component set can have duplicates... but this check is important, and I think I've gotten the logic right with this next commit